src/Configuration/test/ConfigServer.Test/ConfigServerConfigurationProviderTest.cs

```diff
@@ -879,77 +876,86 @@ public void AddConfigServerClientSettings_ChangesDataDictionary()
     {
         var options = new ConfigServerClientOptions
         {
-            AccessTokenUri = "https://foo.bar/",
-            ClientId = "client_id",
-            ClientSecret = "client_secret",
-            Enabled = true,
+            Enabled = false,
+            FailFast = true,
             Environment = "environment",
-            FailFast = false,
             Label = "label",
             Name = "name",
-            Password = "password",
             Uri = "https://foo.bar/",
             Username = "username",
+            Password = "password",
+            Token = "vaultToken",
+            Timeout = 75_000,
+            PollingInterval = TimeSpan.FromSeconds(35.5),
             ValidateCertificates = false,
-            Token = "vaulttoken",
-            TokenRenewRate = 1,
+            AccessTokenUri = "https://token.server.com/",
+            ClientSecret = "client_secret",
+            ClientId = "client_id",
             TokenTtl = 2,
+            TokenRenewRate = 1,
+            DisableTokenRenewal = true,
             Retry =
             {
-                Multiplier = 1.1
+                Enabled = true,
+                InitialInterval = 8,
+                MaxInterval = 16,
+                Multiplier = 1.1,
+                MaxAttempts = 7
+            },
+            Discovery =
+            {
+                Enabled = true,
+                ServiceId = "my-config-server"
+            },
+            Health =
+            {
+                Enabled = false,
+                TimeToLive = 9
+            },
+            Headers =
+            {
+                ["headerName1"] = "headerValue1",
+                ["headerName2"] = "headerValue2"
             }
         };
 
         using var provider = new ConfigServerConfigurationProvider(options, null, null, NullLoggerFactory.Instance);
-        CultureInfo? initialCulture = GetAndSetCurrentCulture(new CultureInfo("ru-RU"));
```

Comment:
It existed to verify the conversion of fractional numbers and dates/times happens in the invariant culture. Without activating a radically different culture, the test would always succeed when run on a US-English system, even when the implementation didn't use the invariant culture, because the culture rules for US-English and invariant are almost identical.

However, there's a huge risk: this test temporarily changes the culture process-wide for _all_ new threads. So when tests are running in parallel (over different assemblies), they may randomly start to fail. Because we have Sonar rules in place these days that warn when the culture isn't specified, I don't think testing for it explicitly is helpful anymore, especially given the randomness in test outcomes it potentially causes.